src/cswinrt/strings/WinRT.cs

```diff
@@ -29,7 +30,12 @@ public static void DynamicInvokeAbi(this System.Delegate del, object[] invoke_pa
     internal sealed class Platform
     {
         [DllImport("api-ms-win-core-com-l1-1-0.dll")]
-        internal static extern unsafe int CoCreateInstance(ref Guid clsid, IntPtr outer, uint clsContext, ref Guid iid, IntPtr* instance);
+        internal static extern unsafe int CoCreateInstance(Guid* clsid, IntPtr outer, uint clsContext, Guid* iid, IntPtr* instance);
+
+        internal static unsafe int CoCreateInstance(ref Guid clsid, IntPtr outer, uint clsContext, ref Guid iid, IntPtr* instance)
+        {
+            return CoCreateInstance((Guid*)Unsafe.AsPointer(ref clsid), outer, clsContext, (Guid*)Unsafe.AsPointer(ref iid), instance);
```

Comment:
Using `Unsafe.AsPointer` like this is a GC hole, and it's not safe.

Reply:
Pinning?

Reply:
Yes, pinning would be the correct way to do things here 🙂

Reply:
Should be pinned now.